hnn_core/network.py

```diff
@@ -219,47 +295,430 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
-    def _add_external_feeds(self):
-        """Legacy function, for backward compatibility with original HNN-GUI
+    def add_evoked_drive(self, name, *, mu, sigma, numspikes, location,
+                         weights_ampa=None, weights_nmda=None,
+                         space_constant=3., synaptic_delays=0.1, seedcore=2):
+        """Add an 'evoked' external drive to the network
+
+        Parameters
+        ----------
+        name : str
+            Unique name for drive
+        mu : float
+            Mean of Gaussian event time distribution (across target cells)
+        sigma : float
+            Standard deviation of event time distribution (across target cells)
+            Set to zero for "synchronous" evoked drive.
+        numspikes : int
+            Number of spikes at each target cell
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        weights_ampa : dict or None
+            Synaptic weights of AMPA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
```

Comment:
is there a difference between zero and omitting the weights?

Reply:
I was trying to express that you can do

```python
weights_ampa = {'L2_pyramidal': 0.1, 'L2_basket': 0.}  # this
weights_ampa = {'L2_pyramidal': 0.1}  # or just this, i.e., those omitted are zero
weights_ampa = None  # or this, i.e., all are zero
```